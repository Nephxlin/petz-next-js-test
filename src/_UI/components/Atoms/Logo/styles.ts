import styled from "styled-components";

interface ClockAnimationProps{
  $showtext?: boolean;
}

export const LogoContainer = styled.div<ClockAnimationProps>`
  display: flex;
  background-color: #E40F0F;
  padding: 13px;
  border-radius: 50px;
  align-items: center;
  cursor: pointer;
  height: 60px;

  ${({ $showtext}) => `
    width: ${ $showtext ? '259px' : '61px'};
  `}
  transition: all 0.5s cubic-bezier(0.250, 0.250, 0.750, 0.750); 

`

export const LogoText = styled.h1<ClockAnimationProps>`
  font-size: 20px;
  font-weight: 600;
  color: #ffffff;
  position: absolute;
  ${({ $showtext}) => `
    margin-left: ${ $showtext ? '50px' : '0px'};
    opacity: ${ $showtext ? 1 : 0};
  `}
  transition: all 0.5s cubic-bezier(0.250, 0.250, 0.750, 0.750); 
`