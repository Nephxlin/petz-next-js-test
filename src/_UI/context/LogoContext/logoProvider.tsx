import { createContext, useContext, useState, useEffect, ReactNode } from 'react';

interface LogoContextType {
  showLogo: boolean;
  setShowLogo: (showLogo: boolean) => void;
}

const LogoContext = createContext<LogoContextType | undefined>(undefined);

interface ClockProviderProps {
  children: ReactNode;
}

export function LogoProvider({ children }: ClockProviderProps) {
  const [showLogo, setShowLogo] = useState<boolean>(true);

  useEffect(() => {
    const intervalId = setInterval(() => {
      setShowLogo(false);
    }, 5000);
    return () => {
      clearInterval(intervalId);
    };
  }, []);

  return (
    <LogoContext.Provider value={{ showLogo, setShowLogo }}>
      {children}
    </LogoContext.Provider>
  );
}

export function useLogo() {
  const context = useContext(LogoContext);
  if (!context) {
    throw new Error('useLogo should be used inside LogoProvider');
  }
  return context;
}
