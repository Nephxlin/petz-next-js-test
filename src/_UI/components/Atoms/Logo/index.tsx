import React, { useEffect, useState } from 'react';
import * as S from './styles'
import Image from "next/image"
import Link from 'next/link';
import { useLogo } from '../../../context/LogoContext/logoProvider';

const Logo = () => {
  const { showLogo } = useLogo();
  const [showText, setShowText] = useState<boolean>(showLogo);
  useEffect(() => {
    setShowText(showLogo)
  }, [showLogo]);
  return (
    <Link href='/'>
      <S.LogoContainer data-testid="logo-container"
        $showtext={showText}
        onMouseEnter={() => setShowText(true)}
        onMouseLeave={() => setShowText(false)}
      >
        <div>
          <Image
            src="/images/white-pokeball.svg"
            alt="Pokebola"
            width={34}
            height={34}
          />
        </div>
        <S.LogoText data-testid='logo-text' $showtext={showText}>Centro Pok√©mon </S.LogoText>
      </S.LogoContainer>
    </Link>
  )
}

export default Logo