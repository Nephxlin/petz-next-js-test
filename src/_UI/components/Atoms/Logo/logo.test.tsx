// write logo render tests here
import { render, screen } from '@testing-library/react';
import Logo from './index';
import { LogoProvider } from '../../../context/LogoContext/logoProvider';
import 'jest-styled-components';

describe('Logo component', () => {
  test('renders Logo component with showLogo as true', () => {
    const { container } = render(
      <LogoProvider>
        <Logo />
      </LogoProvider>
    );
    const logo = screen.getByTestId('logo-container');
    logo.setAttribute('showtext', 'true')
    expect(logo).toBeInTheDocument();
    const logoText = screen.getByText('Centro PokÃ©mon');
    expect(logoText).toBeInTheDocument();
    expect(logoText).toHaveStyleRule('opacity', '1')

    expect(logo).toMatchSnapshot()
  });

  test('renders Logo component with showLogo as false', () => {
    const { container } = render(
      <LogoProvider>
        <Logo />
      </LogoProvider>
    );
    const logo = screen.getByTestId('logo-container');
    const logoText = screen.getByTestId('logo-text');

    setTimeout(() => {
      expect(logoText).toHaveStyleRule('opacity', '0');
      expect(container).toMatchSnapshot();
      done();
    }, 5000);
  });

});

function done() {
  throw new Error('Function not implemented.');
}
