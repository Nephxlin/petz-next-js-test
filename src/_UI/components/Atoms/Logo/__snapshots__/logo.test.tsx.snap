// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component renders Logo component with showLogo as true 1`] = `
.c0 {
  display: flex;
  background-color: #E40F0F;
  padding: 13px;
  border-radius: 50px;
  align-items: center;
  cursor: pointer;
  height: 60px;
  width: 259px;
  transition: all 0.5s cubic-bezier(0.250, 0.250, 0.750, 0.750);
}

.c1 {
  font-size: 20px;
  font-weight: 600;
  color: #ffffff;
  position: absolute;
  margin-left: 50px;
  opacity: 1;
  transition: all 0.5s cubic-bezier(0.250, 0.250, 0.750, 0.750);
}

<div
  class="c0"
  data-testid="logo-container"
  showtext="true"
>
  <div>
    <img
      alt="Pokebola"
      data-nimg="1"
      decoding="async"
      height="34"
      loading="lazy"
      src="/images/white-pokeball.svg"
      srcset="/images/white-pokeball.svg 1x, /images/white-pokeball.svg 2x"
      style="color: transparent;"
      width="34"
    />
  </div>
  <h1
    class="c1"
    data-testid="logo-text"
  >
    Centro Pok√©mon 
  </h1>
</div>
`;
